### ISSUE TITLE
sct_deepseg is _very_ slow

### ISSUE THREAD

**Issue by jcohenadad** [neuropoly/spinalcordtoolbox#3309]
Example below:
```console
sct_deepseg -i t2.nii.gz -task seg_mice_gm-wm_dwi
real	5m15.583s
user	4m43.707s
sys	0m32.558s
```

In comparison, sct_deepseg_sc takes . It would be worth investigating what is the source of the slowdown.
```
time sct_deepseg_sc -i t2.nii.gz -c t2
real	0m5.713s
user	0m11.626s
sys	0m1.802s
```

Note: this is task-dependent. For example, with another task, processing is way faster:
```console
time sct_deepseg -i t2.nii.gz -task seg_sc_t2star
real	0m26.908s
user	0m24.258s
sys	0m4.236s
```

@charleygros is it because the task `seg_mice_gm-wm_dwi` includes a very large upsampling?

**Comment by charleygros** [comment]
> @charleygros is it because the task `seg_mice_gm-wm_dwi` includes a very large upsampling?

Yes, I believe this is exactly why: we are [resampling to 0.1mm](https://github.com/ivadomed/model_seg_mice_gm-wm_dwi_unet2d-multichannel-softseg/blob/main/model_seg_mice_gm-wm_dwi_unet2d-multichannel-softseg.json#L126) in the cross-sectional plane (because the training data was at this resolution).

**Comment by jcohenadad** [comment]
Thank you for your insights @charleygros. I am still wondering if processing time can be improved. More specifically: does the slowness come from the resampling or the DP part?

**Comment by charleygros** [comment]
The DL model is certainly heavier than the other models you mentioned (bigger field of view as input, deeper model in terms of layers), which could slow down the model loading.

I do think that the resampling then 3d Nib reconstruction (ie downsampling) slows the process down.

SOmething else I am considering: use the onnx model (instead of the pt version). I'm adding this to my investigations (w AP flipped data).
